src/github.com/matrix-org/dendrite/roomserver/input/events.go

```diff
@@ -2,12 +2,24 @@ package input
 
 import (
 	"github.com/matrix-org/dendrite/roomserver/api"
+	"github.com/matrix-org/dendrite/roomserver/types"
 	"github.com/matrix-org/gomatrixserverlib"
+	"sort"
 )
 
 // A RoomEventDatabase has the storage APIs needed to store a room event.
 type RoomEventDatabase interface {
-	StoreEvent(event gomatrixserverlib.Event) error
+	// Stores a matrix room event in the database
+	StoreEvent(event gomatrixserverlib.Event, authEventNIDs []int64) error
+	// Lookup the state entries for a list of string event IDs
+	// Returns a sorted list of state entries.
+	StateEntriesForEventIDs(eventIDs []string) ([]types.StateEntry, error)
+	// Lookup the numeric IDs for a list of string event state keys.
+	// Returns a sorted list of state entries.
+	EventStateKeyNIDs(eventStateKeys []string) (map[string]int64, error)
```

Comment:
Also, when is an error returned? Obviously for sql errors, but what if the string cannot be mapped to a nid?